### ISSUE TITLE
TYPO3 9.5 core functional tests fail on parent::setUp() with mssql

### ISSUE THREAD

**Issue by sypets** [TYPO3/testing-framework#433]
I realize that TYPO3 v9 no longer has community support. This is a fyi (for your information). I have removed mssql from my tests.

### What are you trying to achieve?

(1) extension brofix
Execute functional tests for TYPO3 9.5 with typo3/testing-framework to run GitHub CI testing pipelines on TYPO3 extension (for offering support and bugfixes with TYPO3 v9 for this extension). The tests used to work.

(2) core
To double-check, executed functional tests with TYPO3 9.5 branch in core repo.

### What do you get instead?

Tests fail.

Looking at the exception, the failure occurs when calling parent::setUp() 

(1) extension brofix

https://github.com/sypets/brofix/actions/runs/3853332350/jobs/6566224569

(2) core
```
4253) TYPO3\CMS\Workspaces\Tests\Functional\Tca\WorkspaceVisibleFieldsTest::workspaceFormContainsExpectedFields
Doctrine\DBAL\Driver\SQLSrv\SQLSrvException: SQLSTATE [08001, -1]: 
[Microsoft][ODBC Driver 18 for SQL Server]SSL Provider: 
[error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:self signed certificate]
SQLSTATE [08001, -1]: [Microsoft][ODBC Driver 18 for SQL Server]Client unable to establish connection

/var/www/t3coredev9/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvException.php:41
/var/www/t3coredev9/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvConnection.php:50
/var/www/t3coredev9/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/SQLSrv/Driver.php:46
/var/www/t3coredev9/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:362
/var/www/t3coredev9/typo3/sysext/core/Classes/Database/Connection.php:93
/var/www/t3coredev9/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:428
/var/www/t3coredev9/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:388
/var/www/t3coredev9/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:330
/var/www/t3coredev9/typo3/sysext/core/Classes/Database/ConnectionPool.php:197
/var/www/t3coredev9/typo3/sysext/core/Classes/Database/ConnectionPool.php:147
/var/www/t3coredev9/typo3/sysext/core/Classes/Database/ConnectionPool.php:101
/var/www/t3coredev9/typo3/sysext/core/Classes/Preparations/TcaPreparation.php:79
/var/www/t3coredev9/typo3/sysext/core/Classes/Preparations/TcaPreparation.php:45
/var/www/t3coredev9/typo3/sysext/core/Classes/Utility/ExtensionManagementUtility.php:1736
/var/www/t3coredev9/typo3/sysext/core/Classes/Utility/ExtensionManagementUtility.php:1667
/var/www/t3coredev9/typo3/sysext/core/Classes/Core/Bootstrap.php:841
/var/www/t3coredev9/typo3/sysext/core/Classes/Core/Bootstrap.php:145
/var/www/t3coredev9/vendor/typo3/testing-framework/Classes/Core/Testbase.php:628
/var/www/t3coredev9/vendor/typo3/testing-framework/Classes/Core/Functional/FunctionalTestCase.php:334
/var/www/t3coredev9/typo3/sysext/workspaces/Tests/Functional/Tca/WorkspaceVisibleFieldsTest.php:62

```

### How to reproduce the issue?

Use latest TYPO3 9.5 core branch

```
Build/Scripts/runTests.sh -u
Build/Scripts/runTests.sh -s composerInstall -p 7.4
Build/Scripts/runTests.sh -p 7.4 -d mssql -s functional
```

### Specify some data of the environment

- TYPO3 testing framework version: 4.15.5  (see core composer.lock)
- TYPO3 version: 9.5 branch
- TYPO3 installation type: non Composer
- PHP version: 7.4

**Comment by sbuerk** [comment]
mssql container/server comes with with a self signed certificate. the ODBC driver changed the behaviour, using encryption by default. This may be deactivated by a connection argument "Encrypt=no" or "TrustServerCertificate=yes".

Not sure how to add this in TYPO3 for doctrine on connecting to the database. Eventually core changes are needed for that to pass these options in some way. Which is out of the scope for TF and core 9.5. 

Maybe we can configure the "mssql server container" in some way. (at least v10/v11 tests). 

:information_source:  TYPO3 v12 dropped mssql suppport

**Comment by lolli42** [comment]
Thanks for the information and the issue Sybille.

It's good to have this case better documented now for others searching for it. We won't change something in this area anymore, so I hope it's ok to close here, too.

**Comment by sbuerk** [comment]
Some more details now.

With PHP 8.0.27 the ODBC v18 client library is used. In this version, Microsoft
decided to make `Encrypt=Yes` the default for the client library. That means, if
not set on connection the changed default is used which breakds with the MSSQL
server container, which does not havbe SSL configured.

We can mitigate this by setting it as `driverOption` in the testing-framework
functional test:

```php
if ($dbDriver === 'sqlsrv' || $dbDriver === 'pdo_sqlsrv') {
    $localConfiguration['DB']['Connections']['Default']['driverOptions']['Encrypt'] = false;
}
```

TYPO3 v9 is no longer supported, same as TYPO3 v10. TYPO3 v12 removed MSSQL support.
However, TF 6.x and 7.x supports TYPO3 v11 along with MSSQL basically, therefore we
should add this small workaround to mitigate this issue.

That means, we can execute MSSQL tests with runTests.sh in the core for TYPO3 v11 at
least manually. I needed it for a core change, but due to this "issue" MSSQL tests
have been red completly.

I will provide PR's for TF 6 and 7.

**Comment by sbuerk** [comment]
Resolved with #475 and #476.
